controllers/openstack_ansibleee_controller.go

```diff
@@ -75,6 +75,22 @@ func (r *OpenStackAnsibleEEReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	// Initialize Status
+	if instance.Status.Conditions == nil {
+		instance.Status.Conditions = condition.Conditions{}
+
+		cl := condition.CreateList(
+			condition.UnknownCondition(redhatcomv1alpha1.AnsibleExecutionJobReadyCondition, condition.InitReason, redhatcomv1alpha1.AnsibleExecutionJobInitMessage),
+		)
+
+		instance.Status.Conditions.Init(&cl)
+		instance.Status.JobCompleted = false
+
+		// Register overall status immediately to have an early feedback e.g.
+		// in the cli
+		return ctrl.Result{}, nil
+	}
```

Comment:
where do we persist the changes to the instance.Status field? In the rest of the operators there is a deferred function that does that but I don't see it here.

Reply:
as we have only one condition it is set at line 170, when `foundJob.Status.Succeeded`

Reply:
we hsould have other condtions for the input validation or secrete/inventory ectra in the future.

Reply:
@fao89 I don't see where we use the client to send an Update or a Patch request to the kube API server to persist the change of the instance.Status field. Have you tried the current change in a deployment? Does the Status field gets populated in your tests?

Reply:
oh! So that was the thing I was missing! 
I thought I had an issue with my env

Reply:
So for example we have that in keystone https://github.com/openstack-k8s-operators/keystone-operator/blob/b4b2a357b4edfd374510b610201cfc1825df5566/controllers/keystoneapi_controller.go#L127-L139